src/array.rs

```diff
@@ -883,6 +828,53 @@ impl<T: Element, D: Dimension> PyArray<T, D> {
         self.try_readwrite().unwrap()
     }
 
+    fn as_view<S: RawData, F>(&self, from_shape_ptr: F) -> ArrayBase<S, D>
```

Comment:
This is quite a big function, and `from_shape_ptr` is only used in one line.  To avoid code bloat from monomorphization it might be worth splitting out the pieces which don't depend on `from_shape_ptr` into some separate generics.

Reply:
Agreed. I should probably remove the type dependency on `T` as well since that enters only via `itemsize`.